### PR TITLE
Add FAQ on whether restic can resume backups

### PR DESCRIPTION
**Pull request by rawtaz**
What does this PR change? What problem does it solve?
-----------------------------------------------------

This new FAQ entry explains that restic can resume interrupted backups, which previously was not clearly mentioned in the documentation.

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

Clarification needed as evident by discussion in #2960.

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [ ] I have added tests for all changes in this PR
- [x] I have added documentation for the changes (in the manual)
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changelog/TEMPLATE))
- [ ] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: doc/faq.rst [hunk 1]
```diff
@@ -3,6 +3,19 @@ FAQ
...
+Can restic resume an interrupted backup?
+----------------------------------------
+
+Yes it can. When backing up, restic periodically writes index files to keep a record of the uploaded
+data. Even if there's no snapshot created in the end (due to the backup being interrupted), these
+indexes are stored in the database for the data that has been uploaded so far. Next time restic runs,
+it is then able to find the uploaded data through these indexes, and thereby reference it again without
+having to upload it a second time. This effectively makes it continue from where it saved the last index.
+
+It does not matter if the backup was interrupted by the user (e.g. for having to pause it temporarily)
```
Comment (greatroar) [inline doc/faq.rst:15]: I'd leave out the parenthesized part. The user's intention is irrelevant.

**Comment by greatroar** [comment]
I left a nitpicky comment, and I'll leave another: it might be good to explain that restic may rescan files when the snapshot was not completed. When doing the first (presumably interactive) backup of a directory and looking at restic's output, it's immediately clear that it is doing *some* repeated work, but not how much.

Otherwise, good FAQ.

File: doc/faq.rst (hunk 1 above)
Comment (rawtaz) [reply doc/faq.rst]: You're right, less is more.

**Comment by rawtaz** [comment]
I've updated the text, a large part of it actually. Please check again!

File: doc/faq.rst [hunk 2]
```diff
@@ -3,6 +3,25 @@ FAQ
 
 This is the list of Frequently Asked Questions for restic.
 
+Can restic resume an interrupted backup?
+----------------------------------------
+
+Yes it can. When backing up, restic periodically writes index files to keep a record of the uploaded data.
```
Comment (greatroar) [inline doc/faq.rst:9]: Not sure if this was there when I left my previous comment, but I'd rephrase "Yes it can" by something like "It does by default:" (note colon). This is the first bullet point in the FAQ (deservedly so), and a three-word answer comes over as a little snug.

File: doc/faq.rst [hunk 3]
```diff
@@ -3,6 +3,25 @@ FAQ
...
+Even if there's no snapshot created in the end (due to the backup being interrupted), these indexes are
+stored in the repository for the data that has been uploaded so far. Next time restic runs, it is then able
+to find the uploaded data through these indexes, and thereby reference it again without having to upload
+it a second time. This effectively makes it continue from where it saved the last index, which is recently.
```
Comment (greatroar) [inline doc/faq.rst:13]: I'd remove "which is recently".

Comment (rawtaz) [reply doc/faq.rst:13]: If we remove this, it becomes less clear to the user/reader that it's not a long time and thereby won't be a lot of data that's not re-used, I think. At first I wrote "up to 10 minutes", but don't want to put fixed numbers in the documentation. At this point there's nothing in the text that explains how regularly the indexes are created, and with that I feel there is room for them thinking it might be an hour or whatever, it's really undefined. Thoughts?

File: doc/faq.rst (hunk 2 above)
Comment (rawtaz) [reply doc/faq.rst]: I didn't fully agree with your suggestion, but I changed the title from "Can" to "Will" and changes the first sentence.

File: doc/faq.rst (hunk 3 above)
Comment (rawtaz) [reply doc/faq.rst]: Changed the text, see what you think.

Comment (greatroar) [reply doc/faq.rst]: How about replacing "periodically" (first sentence) by "every 10 minutes" and adding a comment in the code (# promised in the FAQ)?

Comment (rawtaz) [reply doc/faq.rst]: The word "periodically" was chosen because it's what fd0 used in the PR adding this functionality to restic, and it's a perfectly valid word. Could be replaced with "regularly", doesn't matter to me.

Replacing it with a fixed number is not good because it adds another data point we have to maintain in case those 10 minutes ever change. We should try to keep documentation as neutral and maintenance-free as possible.

File: doc/faq.rst [hunk 4]
```diff
@@ -3,6 +3,32 @@ FAQ
...
+It does not matter if the backup was interrupted by the user or if it was due
+to unforeseen circumstances such as connectivity issues, power loss, etc.
+Simply re-run the backup again and restic should only upload what it needs to
+in order to complete the interrupted backup and create a snapshot.
+
+Note however that during the initial backup run and any re-tries, until there
+has been a first snapshot created for the backup set (list of files and
+directories to be backed up), restic will need to re-scan the files on disk as
+there is no parent snapshot to compare the filesystem with to determine which
+files has changed. This process should however be far quicker than the
```
Comment (MichaelEischer) [inline doc/faq.rst:28]: s/has/have/

Comment (rawtaz) [reply doc/faq.rst]: Fixed.

**Review by MichaelEischer — APPROVED** [review]
LGTM

File: doc/faq.rst (hunk 3 above)
Comment (greatroar) [reply doc/faq.rst]: As a user, I'd like to have a definite amount of time to wait. As a regular contributor of patches, I'd heed a comment above `repository.indexMagAge = 10 * time.Minute` telling me to `Update the FAQ before changing this constant`. That's all I was trying to say.

Comment (rawtaz) [reply doc/faq.rst]: Yeah, I see. But it makes for a mess. Even if contributors see that, it's still a mess. I'm sorry but it should be enough with what it says now, since "a few minutes" isn't expected to mean something like half an hour or more. But if it's better to remove it that's fine too. I don't want to depend on cross-referencing comments in code though.

Comment (rawtaz) [reply doc/faq.rst]: @greatroar I will go ahead and merge it as it is after having discussed the wording with @MichaelEischer as well. Thanks for your input though, it's always appreciated even if we don't agree fully on specifics all the time :)

File: doc/faq.rst [hunk 5]
```diff
@@ -3,6 +3,32 @@ FAQ
...
+backup being interrupted), these indexes are stored in the repository for the
+data that has been uploaded so far. Next time restic runs, it is then able to
+find the uploaded data through these indexes, and thereby reference it again
+without having to upload it a second time. This effectively makes it continue
+from where it saved the last index, which should be up to a few minutes ago.
+
+It does not matter if the backup was interrupted by the user or if it was due
+to unforeseen circumstances such as connectivity issues, power loss, etc.
+Simply re-run the backup again and restic should only upload what it needs to
+in order to complete the interrupted backup and create a snapshot.
```
Comment (dimejo) [inline doc/faq.rst:22]: IMHO adding a note about `restic recover` would be a good idea in case the user needs to restore files but no snapshot has been created so far.
